libs/cart/state/src/effects/cart-resolver.effects.spec.ts

```diff
@@ -62,278 +55,178 @@ describe('DaffCartResolverEffects', () => {
-          b: cartCreateAction,
-        });
-
-        expect(effects.onResolveCart$).toBeObservable(expected);
+      beforeEach(() => {
+        cartId = stubCart.id.toString();
+        cartStorageService.getCartId.and.returnValue(cartId);
       });
-    });
 
-    describe('when the error thrown is a DaffStorageServiceError', () => {
+      describe('and the get call fails', () => {
+        beforeEach(() => {
+          const response = cold('#', {}, new DaffCartInvalidAPIResponseError('error'));
+          driver.get.and.returnValue(response);
+        });
 
-      it('should indicate that the storage service has failed', () => {
-        cartStorageService.getCartId.and.callFake(throwStorageError)
+        it('should indicate failed cart resolution', () => {
+          const resolveCartFailureAction = new DaffResolveCartFailure('Cart resolution has failed.');
+          const expected = cold('--b', {
+            b: resolveCartFailureAction,
+          });
 
-        actions$ = hot('--a', { a: new DaffResolveCart() });
-        const expected = cold('--b', {
-          b: cartStorageFailureAction
+          expect(effects.onResolveCart$).toBeObservable(expected);
         });
-
-        expect(effects.onResolveCart$).toBeObservable(expected);
       });
-    });
-  });
-
-  describe('onResolveCartSuccess$', () => {
-    describe('when a resolve cart action is followed by a successful cart load', () => {
-      beforeEach(() => {
-        const resolveCartAction = new DaffResolveCart();
-        const loadCartSuccessAction = new DaffCartLoadSuccess(stubCart);
 
-        actions$ = hot('--a--b', {
-          a: resolveCartAction,
-          b: loadCartSuccessAction
+      describe('and the cart is not found when attempting to load it', () => {
+        beforeEach(() => {
+          const response = cold('#', {}, new DaffCartNotFoundError('error'));
+          driver.get.withArgs(cartId).and.returnValue(response);
         });
-      })
 
-      it('should indicate cart resolution success', () => {
-        const resolveCartSuccessAction = new DaffResolveCartSuccess();
-        const expected = cold('-----a', {a: resolveCartSuccessAction});
-
-        expect(effects.onResolveCartSuccess$).toBeObservable(expected);
-      })
-    })
-
-    describe('when a resolve cart action is followed by a successful cart creation', () => {
-      beforeEach(() => {
-        const resolveCartAction = new DaffResolveCart();
-        const createCartSuccessAction = new DaffCartCreateSuccess(stubCart);
-
-        actions$ = hot('--a--b', {
-          a: resolveCartAction,
-          b: createCartSuccessAction
+        describe('and the driver calls succeed', () => {
+          beforeEach(() => {
+            const newCartId = 'newCartId';
+            driver.create.and.returnValue(of({id: newCartId}));
+            driver.get.withArgs(newCartId).and.returnValue(of(stubCart));
+          });
+
+          it('should create a new cart', () => {
```

Comment:
I got a little confused when reading through these last set of `and the cart is not found when attempting to load it` tests. I think it is because this test should be moved up one level, since it happens whether the driver calls are successful or not.

Reply:
moved up a level